GenericList/list.h

```diff
@@ -0,0 +1,240 @@
+#pragma once
+
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <memory>
+#include <cassert>
+#include <functional>
+#include <algorithm>
+
+using namespace std;
+
+
+template <typename T>
```

Comment:
You can make the allocator a template as well. `std::allocator` is stateless, but there are allocators the are stateful how would you handle this during moves/copies?
```c++
template<typename T, typename Alloc = std::allocator<T>>
class List {
T* data;
size_t capacity;
size_t count;
Alloc allocator;
public:
List() : data(), capacity(), count() {}
List(Alloc const& alloc) : data(), capacity(), count(), allocator(alloc) {}
...
};
```